src/multi_physics/QED/src/physics/quantum_sync/quantum_sync_engine_tables.hpp

```diff
@@ -0,0 +1,779 @@
+                                           default_chi_part_max<RealType>,
+                                           default_chi_part_how_many};
+
+    /**
+    * generation_policy::force_internal_double can be used to force the
+    * calculations of a lookup tables using double precision, even if
+    * the final result is stored in a single precision variable.
+    */
+    enum class generation_policy{
+        regular,
+        force_internal_double
+    };
+
+    /**
+    * This class provides the lookup table for dN/dt,
+    * storing the values of the G function (see validation script)
+    * and providing methods to perform interpolations.
+    * It also provides methods for serialization (export to byte array,
+    * import from byte array) and to generate "table views" based on
+    * non-owning poynters (this is crucial in order to use the table
+    * in GPU kernels, as explained below).
+    *
+    * Internally, this table stores log(G(log(chi))).
+    *
+    * @tparam RealType the floating point type to be used
+    * @tparam VectorType the vector type to be used internally (e.g. std::vector)
+    */
+    template<
+        typename RealType,
+        typename VectorType>
+    class dndt_lookup_table
+    {
+        public:
+
+            /**
+            * A view_type is essentially a dN/dt lookup table which
+            * uses non-owning, constant, pointers to hold the data.
+            * The use of views is crucial for GPUs. As demonstrated
+            * in test_gpu/test_quantum_sync.cu, it is possible to:
+            * - define a thin wrapper around a thrust::device_vector
+            * - use this wrapper as a template parameter for a dndt_lookup_table
+            * - initialize the lookup table on the CPU
+            * - generate a view
+            * - pass by copy this view to a GPU kernel (see get_view() method)
+            *
+            * @tparam RealType the floating point type to be used
+            */
+            typedef const dndt_lookup_table<
+                RealType, containers::picsar_span<const RealType>> view_type;
+
+            /**
+            * Constructor (not designed for GPU usage)
+            * After construction the table is empty. The user has to generate
+            * the G function values before being able to use the table.
+            *
+            * @param params table parameters
+            */
+            dndt_lookup_table(dndt_lookup_table_params<RealType> params):
+            m_params{params},
+            m_table{containers::equispaced_1d_table<RealType, VectorType>{
+                    math::m_log(params.chi_part_min),
+                    math::m_log(params.chi_part_max),
+                    VectorType(params.chi_part_how_many)}}
+            {};
+
+            /**
+            * Constructor (not designed for GPU usage)
+            * After construction the table is empty. The user has to generate
```

Comment:
Same here as for breit-wheeler, is the table actually empty? (same below for the 2D table)

Reply:
"uninitialized" is more correct indeed. I've amended the comment.